source/core.py

```diff
@@ -250,6 +250,48 @@ def onEndSession(evt):
 	import windowUtils
 	class MessageWindow(windowUtils.CustomWindow):
 		className = u"wxWindowClassNR"
+		batteryCache = None
+		REASON_INITIALIZE = 1
+		def __init__(self, windowName=None):
+			super(MessageWindow, self).__init__(windowName)
+			self.say_battery_status(reason=self.REASON_INITIALIZE)
+	
+		def windowProc(self, hwnd, msg, wParam, lParam):
+			import ui
+			#Just define these constants here, so we don't have to import win32con
+			WM_POWERBROADCAST = 0x218
+			WM_DISPLAYCHANGE = 0x7e
+			PBT_APMPOWERSTATUSCHANGE = 0xA
+			if msg == WM_POWERBROADCAST and wParam == PBT_APMPOWERSTATUSCHANGE:
+				self.say_battery_status()
+			# Resolution detection comes from an article found at https://msdn.microsoft.com/en-us/library/ms812142.aspx.
+			elif msg == WM_DISPLAYCHANGE:
+				ui.message("Landscape" if lParam%0x10000 > lParam/0x10000 else "Portrait")
+
+
+		def say_battery_status(self, reason=None):
+			#Mostly taken from script_say_battery_status, but modified.
+			import ui
+			import winKernel
+			UNKNOWN_BATTERY_STATUS = 0xFF
+			AC_ONLINE = 0X1
+			NO_SYSTEM_BATTERY = 0X80
+			sps = winKernel.SYSTEM_POWER_STATUS()
+			if not winKernel.GetSystemPowerStatus(sps) or sps.BatteryFlag is UNKNOWN_BATTERY_STATUS:
+				return
+			if sps.BatteryFlag & NO_SYSTEM_BATTERY or sps.ACLineStatus == self.batteryCache:
+				return
+			#Necessary because sometimes, this event double fires, and also, if the battery decreases by 3 percent, this event occurs.

```

Comment:
Nope.


On 9/4/2016 11:51 PM, James Teh wrote:
>
> In source/core.py 
> <https://github.com/nvaccess/nvda/pull/6320#discussion_r77472714>:
>
> > +				ui.message("Landscape" if lParam%0x10000 > lParam/0x10000 else "Portrait")
> > +
> > +
> > +		def say_battery_status(self, reason=None):
> > +			#Mostly taken from script_say_battery_status, but modified.
> > +			import ui
> > +			import winKernel
> > +			UNKNOWN_BATTERY_STATUS = 0xFF
> > +			AC_ONLINE = 0X1
> > +			NO_SYSTEM_BATTERY = 0X80
> > +			sps = winKernel.SYSTEM_POWER_STATUS()
> > +			if not winKernel.GetSystemPowerStatus(sps) or sps.BatteryFlag is UNKNOWN_BATTERY_STATUS:
> > +				return
> > +			if sps.BatteryFlag & NO_SYSTEM_BATTERY or sps.ACLineStatus == self.batteryCache:
> > +				return
> > +			#Necessary because sometimes, this event double fires, and also, if the battery decreases by 3 percent, this event occurs.
>
> Do you mean "decreases to 3 percent", rather than "by 3 percent"?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/nvaccess/nvda/pull/6320/files/69b19519a04ca0f5c647bb9023d1eb3d8c9f5e47#r77472714>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AFGivQT70FERf51UY9dEUcT6uDFfhItIks5qm63_gaJpZM4JxF3F>.
>

-- 
------------------------------------------------------------------------


    Derek Riemer

  * Department of computer science, third year undergraduate student.
  * Proud user of the NVDA screen reader.
  * Open source enthusiast.
  * Member of Bridge Cu
  * Avid skiier.

Websites:
Honors portfolio <http://derekriemer.com>
Awesome little hand built weather app! 
<http://django.derekriemer.com/weather/>

email me at derek.riemer@colorado.edu <mailto:derek.riemer@colorado.edu>
Phone: (303) 906-2194